src/TimeZones.jl

```diff
@@ -33,13 +33,16 @@ export TimeZone, @tz_str, istimezone, FixedTimeZone, VariableTimeZone, ZonedDate
 
 _scratch_dir() = @get_scratch!("build")
 
-const _COMPILED_DIR = Ref{String}(TZJData.ARTIFACT_DIR)
+const _COMPILED_DIR = Ref{String}()
 
 # TimeZone types used to disambiguate the context of a DateTime
 # abstract type UTC <: TimeZone end  # Already defined in the Dates stdlib
 abstract type Local <: TimeZone end
 
 function __init__()
+    # Must be set at runtime to ensure relocatability 
+    _COMPILED_DIR[] = TZJData.artifact_dir()
```

Comment:
A option for backwards compat:
```suggestion
    # Backwards compatibility with TZJData v1.3.0 and below. Using older versions means TimeZones.jl is not relocatable
    _COMPILED_DIR[] = isdefined(TZJData, :artifact_dir) ? TZJData.artifact_dir() : TZJData.ARTIFACT_DIR
```

Reply:
If we want to ensure backwards compatibility then relocatability can only be enforced by adding a lower bound to the `TZJData` version. I would see adding a lower bound to the version of `TimeZones` as a more natural way to do it, since that's the package I expect people usually add to their `Project.toml`.
Unless there is sometimes the need for keeping old versions of `TZJData` pinned, in which case backwards compatibility becomes really necessary. I don't know the package well enough to say if that's the case.

Reply:
Yeah, I think most downstream users are depending on TimeZones.jl, not TZJData.jl.

Is there a pressing need for TimeZones.jl to support old versions of TZJData.jl? If so, could we just backport https://github.com/JuliaTime/TZJData.jl/pull/32 to older versions of TZJData.jl?

Reply:
Part of the reason for the TimeZones.jl and TZJData.jl package separation is to allow for changing tzdata independently from the TimeZones.jl code. Setting a lower bound to TZJData.jl means we can't use those old versions. That's not the end of the world but the suggestion I had allows for relocatability and compatibility.

Backporting https://github.com/JuliaTime/TZJData.jl/pull/32 is possible but we would still need to update the compat in TimeZones.jl to allow for each of those patch releases.

Reply:
I have added a fallback for older versions by hardcoding the artifact hashes and querying them using `TZDATA_VERSION`. This is not very elegant but 1) there is no 100% foolproof solution to retrieve the hash AFAIK 2) there are not too many versions. In this way we can ensure relocatability even with the old TZJDatas.

Reply:
Do you care about relocatability for old TZJData versions? I'm fine with using `ARTIFACT_DIR` as the fallback. I'd rather break relocatability for old versions than break compat support unnecessarily (if that is a problem).

Reply:
I think the current fallback works with all old versions of TZJData, so there is no need to change the `[compat]` bounds.

Reply:
I have tested it for the last released version of TZJData on a system image and it works, even without the relocatability PR for TZJData.